v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/model/Mod.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub.model;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.bigtable.data.v2.models.ChangeStreamMutation;
+import com.google.cloud.bigtable.data.v2.models.ChangeStreamMutation.MutationType;
+import com.google.cloud.bigtable.data.v2.models.DeleteCells;
+import com.google.cloud.bigtable.data.v2.models.DeleteFamily;
+import com.google.cloud.bigtable.data.v2.models.Range.BoundType;
+import com.google.cloud.bigtable.data.v2.models.SetCell;
+import com.google.common.collect.Maps;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.io.Serializable;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Base64;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.beam.sdk.coders.AvroCoder;
+import org.apache.beam.sdk.coders.DefaultCoder;
+import org.apache.commons.lang3.StringUtils;
+import org.threeten.bp.Instant;
+
+/**
+ * The {@link Mod} contains the keys, new values (from {@link
+ * com.google.cloud.bigtable.data.v2.models.Entry}) and metadata ({@link
+ * com.google.cloud.bigtable.data.v2.models.ChangeStreamMutation}) of a Bigtable changelog row.
+ */
+@DefaultCoder(AvroCoder.class)
+public final class Mod implements Serializable {
+
+  private static final long serialVersionUID = 8703757194338184299L;
+
+  private static final String PATTERN_FORMAT = "yyyy-MM-dd HH:mm:ss.SSSSSS";
+  private static final ThreadLocal<DateTimeFormatter> TIMESTAMP_FORMATTER =
+      ThreadLocal.withInitial(
```

Comment:
Since we use numeric timestamps this will eventually become unused if we delete all references to textual timestamps and functions which produce them